irr/include/vector2d.h

```diff
@@ -34,6 +34,11 @@ class vector2d
 	constexpr vector2d(const dimension2d<T> &other) :
 			X(other.Width), Y(other.Height) {}
 
+	template <class U>
+	constexpr vector2d<U> cast() const {
+		return vector2d<U>(static_cast<U>(X), static_cast<U>(Y));
+	}
```

Comment:
:+1: for idea but maybe
`template <class U> constexpr static vector2d<T> from(const vector2d<U> &other)`
would make more sense conceptually.
`a.cast<f32>()` vs. `v3f::from(a)`

Reply:
done